http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -414,53 +458,140 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
       )
 
+      // Invoke helm
+      _ <- helmClient
+        .installChart(
+          org.broadinstitute.dsp.Release(releaseName.value),
+          org.broadinstitute.dsp.Chart(config.galaxyAppConfig.chart.value),
+          org.broadinstitute.dsp.Values(chartValues)
+        )
+        .run(helmAuthContext)
+
+      // Poll galaxy until it starts up
+      // TODO potentially add other status checks for pod readiness, beyond just HTTP polling the galaxy-web service
+      isDone <- streamFUntilDone(galaxyDAO.isProxyAvailable(dbCluster.googleProject, appName),
+                                 config.monitorConfig.createApp.maxAttempts,
+                                 config.monitorConfig.createApp.interval).compile.lastOrError
+
+      _ <- if (!isDone) {
+        val msg =
+          s"Galaxy installation has failed or timed out for app ${appName.value} in cluster ${dbCluster.getGkeClusterId.toString} | trace id: ${ctx.traceId}"
+        logger.error(msg) >>
+          F.raiseError[Unit](AppCreationException(msg))
+      } else F.unit
+
+    } yield ()
+
+  private[util] def uninstallGalaxy(dbCluster: KubernetesCluster,
+                                    appName: AppName,
+                                    namespaceName: NamespaceName,
+                                    googleCluster: Cluster)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- ev.ask
+
+      _ <- logger.info(
+        s"Uninstalling helm chart ${config.galaxyAppConfig.chart.value} for app ${appName.value} in cluster ${dbCluster.getGkeClusterId.toString} | trace id: ${ctx.traceId}"
+      )
+
+      helmAuthContext <- getHelmAuthContext(googleCluster, dbCluster.id, namespaceName)
+
+      releaseName = buildReleaseName(appName)
+
+      // Invoke helm
+      _ <- helmClient
+        .uninstall(org.broadinstitute.dsp.Release(releaseName.value), config.galaxyAppConfig.uninstallKeepHistory)
+        .run(helmAuthContext)
+
+      last <- streamFUntilDone(
+        kubeService.listPodStatus(dbCluster.getGkeClusterId, KubernetesNamespace(namespaceName)),
+        config.monitorConfig.deleteApp.maxAttempts,
+        config.monitorConfig.deleteApp.interval
+      ).compile.lastOrError
+
+      _ <- if (!podDoneCheckable.isDone(last)) {
+        val msg =
+          s"Galaxy deletion has failed or timed out for app ${appName.value} in cluster ${dbCluster.getGkeClusterId.toString}. The following pods are not in a terminal state: ${last
+            .filterNot(isPodDone)
+            .map(_.name.value)
+            .mkString(", ")} | trace id: ${ctx.traceId}"
+        logger.error(msg)
+
+        // TODO (RT): currently there seems to be a bug causing Galaxy pods to not terminate when `helm delete` is called.
+        // So we're currently _not_ throwing an error if deletion times out, but we should uncomment the below
+        // line once the Galaxy bug is fixed.
```

Comment:
Do we know if those `Terminating` pods are removed once the namespace is deleted?

Reply:
They are not - the namespace itself gets stuck in Terminating and the pods also stay in Terminating.

It sounded like Alex had an idea about what the issue is.